ref/general/application-hardening.adoc

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2020 IBM Corporation and others.
+// Licensed under Creative Commons Attribution-NoDerivatives
+// 4.0 International (CC BY-ND 4.0)
+//   https://creativecommons.org/licenses/by-nd/4.0/
+//
+// Contributors:
+//     IBM Corporation
+//
+:page-description: If your application isn't properly secured, external users might derive or inherit unauthorized privileges when they run it. Before you can run an application in production, you must carefully harden its configuration.
+:seo-title: Application configuration security hardening
+:seo-description: If your application isn't properly secured, external users might derive or inherit unauthorized privileges when they run it. Before you can run an application in production, you must carefully harden its configuration.
+:page-layout: general-reference
+:page-type: general
+= Application configuration security hardening
+
+If your application isn't properly secured, external users might derive or inherit unauthorized privileges when they run it.
+This unauthorized access can lead to users with access to sensitive information about your application and other users.
+Before you can run an application in production, you must carefully harden its configuration.
+
+== User roles and access
+Understanding the roles and mappings of users is key to securely configuring your application for production use.
+For information about how to map users to roles and groups, see link:/docs/ref/general/#authorization.html[Authorization].
+
+If you don't configure user roles in the `server.xml` file, Open Liberty uses the group name that the application runs under as the role name by default.
+For conflicts in a servlet's role mapping between an application's `web.xml` file and annotations in the application, the config in the `web.xml` file takes precedence.
+If there is a conflict between the `ibm-application-bnd.xmi` and `server.xml` files, then the config int he `server.xml` file takes precedence.
+
+When the role mapping binding information for a protected application isn't provided, the group name and role name must be the same.
+For example, if the role name is `manager`, then a user who belongs to a `manager` group has access to that resource.
+This applies only when application binding information isn't specified for the application in the `ibm-application-bnd.xmi` or `server.xml` files
```

Comment:
I think maybe add a noun after "This" here to avoid any possible confusion. So maybe change it to "This example"

Reply:
Updated.